File: user-service/src/main/java/rs/raf/user_service/entity/Company.java
package rs.raf.user_service.entity;

import javax.persistence.*;

@Entity
public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @Column(unique = true)
    private String registrationNumber;
    @Column(unique = true)
    private String taxId;
    private Integer activityCode; // sifra placanja
    private String address;
    @ManyToOne
    private Client majorityOwner;
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/entities/Company.kt
package pack.userservicekotlin.domain.entities

import jakarta.persistence.*

@Entity
open class Company(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null,
    open var name: String? = null,
    @Column(updatable = false, unique = true)
    open var registrationNumber: String? = null,
    @Column(updatable = false, unique = true)
    open var taxId: String? = null,
    open var activityCode: String? = null,
    open var address: String? = null,
    @ManyToOne
    open var majorityOwner: Client? = null,
    @OneToMany(mappedBy = "company", cascade = [CascadeType.ALL], orphanRemoval = true)
    open var authorizedPersonel: MutableList<AuthorizedPersonel> = mutableListOf(),
)
